### PR TITLE
Run cmds

### PR DESCRIPTION
**Pull request by computerlyrik**
Refs https://github.com/rhuss/docker-maven-plugin/issues/191
@rhuss/docker-maven-plugin#191

### CODE REVIEW

File: src/main/java/org/jolokia/docker/maven/assembly/DockerFileDictionaryEnum.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package org.jolokia.docker.maven.assembly;
+
+/**
+ * @author  Paris Apostolopoulos <javapapo@mac.com></javapapo@mac.com> on 6/13/15.
+ */
+public enum DockerFileDictionaryEnum {
```
Comment (tnn) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerFileDictionaryEnum.java]: No need to postfix the enum with Enum.

File: src/main/java/org/jolokia/docker/maven/assembly/DockerAssemblyManager.java [hunk 2]
```diff
@@ -36,7 +39,9 @@
 @Component(role = DockerAssemblyManager.class)
 public class DockerAssemblyManager {
 
-    public static final String DEFAULT_DATA_BASE_IMAGE = "busybox:latest";;
+    public static final String DEFAULT_DATA_BASE_IMAGE = "busybox:latest";
+
+    ;
```
Comment (tnn) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerAssemblyManager.java]: I suggest removing the redundant `;`

File: src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java [hunk 3]
```diff
@@ -179,6 +186,14 @@ private void addPorts(StringBuilder b) {
         }
     }
 
+    private void addRunCmds(StringBuilder b) {
+        if (runcmds.size() > 0) {
```
Comment (tnn) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java]: This is redundant, for each wouldn't run if size is 0.

File: src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java [hunk 4]
```diff
@@ -264,6 +279,22 @@ public DockerFileBuilder expose(List<String> ports) {
         return this;
     }
 
+    /**
+     * Adds the RUN Commands within the build image section
+     * @param runCmds
+     * @return
+     */
+    public DockerFileBuilder runCommands(List<String> runCmds) {
```
Comment (tnn) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java]: I would probably put runCmd != null as invariant and guard with `Objects.requireNotNull(runCmds)`.

File: src/main/java/org/jolokia/docker/maven/assembly/DockerFileDictionaryEnum.java [hunk 5]
```diff
@@ -0,0 +1,13 @@
+package org.jolokia.docker.maven.assembly;
+
+/**
+ * @author  Paris Apostolopoulos <javapapo@mac.com></javapapo@mac.com> on 6/13/15.
```
Comment (tnn) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerFileDictionaryEnum.java]: Git blame will remember the author, also you put the email as a xml tag (IntelliJ autocomplete I assume). :)

File: src/main/java/org/jolokia/docker/maven/config/BuildImageConfiguration.java [hunk 6]
```diff
@@ -132,6 +139,13 @@ public Arguments getEntryPoint() {
         return entryPoint;
     }
 
+    public List<String> getRunCmds() {
+        return runCmds;
+    }
+
+    public void setRunCmds(List<String> runCmds) {
```
Comment (tnn) [inline src/main/java/org/jolokia/docker/maven/config/BuildImageConfiguration.java]: The build image configuration is immutable, so it shouldn't allow setters.

File: src/main/java/org/jolokia/docker/maven/config/BuildImageConfiguration.java [hunk 7]
```diff
@@ -165,6 +179,11 @@ public Builder ports(List<String> ports) {
             config.ports = ports;
             return this;
         }
+
+        public Builder runCmds(List<String> theCmds) {
+            config.runCmds = theCmds;
```
Comment (tnn) [inline src/main/java/org/jolokia/docker/maven/config/BuildImageConfiguration.java]: I would guard against null here.

File: src/test/java/org/jolokia/docker/maven/assembly/DockerFileBuilderTest.java [hunk 8]
```diff
@@ -26,68 +25,89 @@ public void testBuildDockerFile() throws Exception {
...
+				dockerfileToMap(dockerfileContent),
+				hasEntry("ENTRYPOINT",
+						"[\"java\",\"-jar\",\"/my-app-1.1.1.jar\",\"server\"]"));
+	}
 
-    @Test
-    public void testNoRootExport() {
-        assertFalse(new DockerFileBuilder().add("/src", "/dest").basedir("/").content().contains("VOLUME"));
-    }
+	@Test
```
Comment (tnn) [inline src/test/java/org/jolokia/docker/maven/assembly/DockerFileBuilderTest.java]: Please do reformatting in a separate PR or at least separate commit.

File: src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java (hunk 4 above)
Comment (computerlyrik) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java]: But runCmds could be null? As ports do also - what else is the difference to ports (handled equally in your source code)

**Comment by computerlyrik** [comment]
Fixed the issues , i think

File: src/test/java/org/jolokia/docker/maven/assembly/DockerFileBuilderTest.java (hunk 8 above)
Comment (computerlyrik) [inline src/test/java/org/jolokia/docker/maven/assembly/DockerFileBuilderTest.java]: i am sorry about this, but it is code from @javapapo 
i only did some work ahead of it to push it to a pull request
Please reformat as needed. Thank you.

File: src/main/java/org/jolokia/docker/maven/assembly/DockerAssemblyManager.java [hunk 9]
```diff
@@ -147,7 +154,8 @@ private File createTarball(BuildDirs buildDirs, File extraDir, AssemblyMode buil
         }
     }
 
-    private TarArchiver createArchiver(File outputDir, File archive, AssemblyMode buildMode) throws NoSuchArchiverException {
+    private TarArchiver createArchiver(File outputDir, File archive, AssemblyMode buildMode) throws
```
Comment (osigida) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerAssemblyManager.java]: why have you changed formatting?

File: src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java (hunk 4 above)
Comment (tnn) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java]: Would that be in the case of 
```xml
<runCommands>
</runCommands>
```
?

File: src/main/java/org/jolokia/docker/maven/config/handler/property/PropertyConfigHandler.java [hunk 10]
```diff
@@ -138,6 +139,19 @@ private String extractName(String prefix, Properties properties) throws IllegalA
...
+
+    private List<String> extractRunCommands(String prefix, Properties properties) {
+        List<String> ret = new ArrayList<>();
+        List<String> cmds = listWithPrefix(prefix, RUNCMDS, properties);
+        if (cmds == null) {
+            return null;
+        }
+//        for (String[] port : parsedPorts) {
+//            ret.add(port[1]);
+//        }
```
Comment (osigida) [inline src/main/java/org/jolokia/docker/maven/config/handler/property/PropertyConfigHandler.java]: please delete commented code

File: src/test/java/org/jolokia/docker/maven/assembly/DockerFileBuilderTest.java [hunk 11]
```diff
@@ -11,10 +11,9 @@
 import static org.hamcrest.Matchers.hasEntry;
 import static org.junit.Assert.*;
 
-
 public class DockerFileBuilderTest {
 
-    @Test
+	@Test
```
Comment (osigida) [inline src/test/java/org/jolokia/docker/maven/assembly/DockerFileBuilderTest.java]: tab instead on spaces?

File: src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java (hunk 4 above)
Comment (tnn) [inline src/main/java/org/jolokia/docker/maven/assembly/DockerFileBuilder.java]: Please also note, rhuss in the owner - I'm just a casual user, who would love see RUN support. :)

File: src/test/java/org/jolokia/docker/maven/assembly/DockerFileBuilderTest.java (hunk 8 above)
Comment (javapapo) [inline src/test/java/org/jolokia/docker/maven/assembly/DockerFileBuilderTest.java]: Sorry for that if it was me, tried not to change the formatting, must have missed something.

File: src/main/java/org/jolokia/docker/maven/config/handler/property/PropertyConfigHandler.java (hunk 10 above)
Comment (computerlyrik) [inline src/main/java/org/jolokia/docker/maven/config/handler/property/PropertyConfigHandler.java]: done

**Comment by rhuss** [comment]
Thanks a lot ! I will integrate it for 0.13.2 (probably next week).

In the meantime could you do me a favor and sign-off your PR ? How this is done is explained [here](https://github.com/rhuss/docker-maven-plugin/blob/master/doc/contributing.md). Also, you could rebase the commits to a single commit (rebase -i), which would makes things easier. This is not absolutely necessary, but let me know if you want to do this before I start to integrate.

Thanks again !

**Comment by computerlyrik** [comment]
Hi! Thanks for the positive feedback.

I tried my best with git, 
- rebased into single commit
- tried to exclude the code-formattings from @javapapo 
- merged against your master 0.13.2-SNAPSHOT

but this result isn't completely what i expected.

Perhaps you just want to cherry-pick this commit: 
https://github.com/computerlyrik/docker-maven-plugin/commit/720af591925fd185e5988a5f22ccf0c1ab32a21e

**Comment by computerlyrik** [comment]
Closed in Favor of this

https://github.com/rhuss/docker-maven-plugin/pull/218
